Utils/Functions/Apps/Resources/script.js

```diff
@@ -1,36 +1,41 @@
 /**
-* AppResource prototype
-* @constructor
-*/
-function AppResource() {
-    this._appsManager = Bean("repositoryManager");
-}
+ * AppResource class
+ */
+// eslint-disable-next-line no-unused-vars
+class AppResource {
+    constructor() {
+        this.appsManager = Bean("repositoryManager");
+    }
 
-/**
-* sets application
-* @param {string} application application of the resource
-* @returns {AppResource} AppResource object
-*/
-AppResource.prototype.application = function (application) {
-    this._application = application;
-    return this;
-}
+    /**
+     * Sets the application containing the resources
+     *
+     * @param {string} application The application with the resource
+     * @returns {AppResource} The AppResource object
+     */
+    application(application) {
+        this._application = this.appsManager.getApplication(application);
```

Comment:
What if `getApplication` does not work? Does it throw an exception internally?

Reply:
What do you mean by does not work? If it returns `null` then https://github.com/PhoenicisOrg/scripts/pull/1046/files#diff-a8f2ab0b620d11b4b57a6e3699a24369R29 should trigger.
If it throws an exception then the exception should be shown inside an `ErrorDialog`

Reply:
Wouldn't it make sense to stop already here when it returns `null` (fail early)?

Reply:
I don't know. In theory this is simply a setter, so the script using `AppResource` would still be able to provide a valid application.

Reply:
Only print a warning then?

Reply:
If we assume that the script using the `AppResource` has no access to the created `_application` field I agree, otherwise we shouldn't print a warning in my opinion.

Reply:
Do we assume that `_application` is private?

Reply:
Yes.

Reply:
Done